cypress/lib/util.ts

```diff
@@ -76,11 +80,54 @@ export const fakeLogin = (subscriber = true) => {
 		'https://members-data-api.theguardian.com/user-attributes/me',
 		response,
 	).as('userData');
-}
+};
 
- export const fakeLogOut = () => {
+export const fakeLogOut = () => {
 	// we can't just click sign out because cypress does not like following links to other domains
 	cy.clearCookie('GU_U');
 
 	cy.reload();
-}
+};
+
+export const mockIntersectionObserver = (win: Window, selector?: string) => {
+	const Original = win.IntersectionObserver;
+	// ts-expect-error
+	win.IntersectionObserver = function (
```

Comment:
Relying on scrolling seems to be incredibly unreliable, there were all sorts of weird problems with it, resulting in it often scrolling to the wrong place and getting stuck, therefore not actually triggering the lazy loading.

Setting the scroll duration helped somewhat for some tests, but similar to `wait` it's a bit of an anti-pattern and didn't always work.

We could check for requests to GAM instead of an iframe being inserted, but we still need to trigger the lazy loading first anyway.

I'm comfortable with this as we're not testing the lazy loading itself which we could add as another test separate test.

Reply:
Yep scrolling can be tricky. The `hydrate` command uses scrolling for islands and that seems to work okay though. 

Happy to keep as is for now and we can continue to brainstorm other methods.